### DISCUSSION TITLE
How are you using wire:model with blade components?

### DISCUSSION THREAD

**Discussion by AdamKyle** [livewire/livewire#1855] [4. General]
I was watching a screen cast that the author made for data-tables, and he uses a lot of blade components. `<x-whatever />` which I started to do, but then I had questions: **How are you using wire:model on a blade component?**

For example I created:

```
<div class="col-md-8 form-inline">
    Per Page: &nbsp;
    <select class="form-control">
        <option>10</option>
        <option>15</option>
        <option>25</option>
    </select>
</div>
```

Thats called as such: `<x-data-tables::per-page wire:model="perPage" />`

The component renders fine with in the livewire component, but selecting anything from the list does not update the table. I think its because it's like "what exactly am I doing?" 

The author did show their components, but I didn't see any `wire:model=""` deep with in the blade components, he applied it to the top level `<x-whatever />` and went about his day.

Am I missing something? How do you separate your components out but still be able to use them with livewire? I don't see that documented any where. Ideas?

**Comment by iksaku** [comment]
While using Blade Components, you need to make sure to [Properly Manage your Attributes](https://laravel.com/docs/8.x/blade#managing-attributes).

`wire:model` is a custom HTML attribute used by Livewire, and it needs to be passed down in your Blade component, which at the moment I see you're not doing.

The most basic way to solve your issue is by passing the `$attributes` blade property in your input:
```blade
<div class="col-md-8 form-inline">
    Per Page: &nbsp;
    <select {{ $attributes->merge(['class' => 'form-control']) }}>
        <option>10</option>
        <option>15</option>
        <option>25</option>
    </select>
</div>
```

This will allow your Blade component to properly grab the `wire:model` directive you're willing to use, as well as to open the gate for you to add any other attribute if needed.
